setup.py

```diff
@@ -106,10 +106,11 @@
     'ipython>=4.0.0',
     'ipykernel>=4.2.2',
     'traitlets>=4.2.0',
-    'widgetsnbextension>=1.0.0b4',   # TODO: Remove this dependency in 6.0
-                                     # (We should be able to install the python
-                                     #  side of ipywidgets on bare python
-                                     #  kernels)
+    # TODO: Dynamically add this dependency
+    # only if notebook 4.x is installed in this
+    # interpreter, to allow ipywidgets to be
+    # installed on bare kernels.
+    'widgetsnbextension>=1.0.0b4',
```

Comment:
So a path forward for the dependency of ipywidgets to widgetsnbextension is the following.
 - you can install ipywidgets on a bare kernel. If notebook 4.x is not there, don't add the widgetsnbextension.
 - If notebook 4.x is installed on this interpreter, it does not hurt to install widgetsnbextension and we still keep the one-step install for standalone users.